### PR TITLE
Arc 2643 audit log table entry

### PR DESCRIPTION
**Pull request by kamaksheeAtl**
**What's in this PR?**
Complete audit log entry for build workflow run calls

**Added feature flags**
use-dynamodb-to-persist-audit-log

**How has this been tested?**  
locally and via unit test cases

**Whats Next?**
work on audit log entry for deployments

### CODE REVIEW

File: src/github/workflow.ts [hunk 1]
```diff
@@ -30,8 +31,18 @@ export const workflowWebhookHandler = async (context: WebhookContext, jiraClient
...
 	logger.info({ jiraHost: jiraClient.baseURL }, `Sending workflow event to Jira`);
+	const subscription = await Subscription.getSingleInstallation(jiraHost, gitHubInstallationId, gitHubAppId);
+	if (!subscription) {
+		logger.warn(`Subscription not found to log the info into audit log table`);
+	}
 
-	const jiraResponse = await jiraClient.workflow.submit(jiraPayload, payload.repository.id);
+	const jiraResponse = await jiraClient.workflow.submit(jiraPayload, payload.repository.id, {
+		preventTransitions: true,
+		operationType: "BACKFILL",
```
Comment (gxueatlassian) [inline src/github/workflow.ts:41]: This should be `NORMAL` right?

File: src/github/workflow.ts [hunk 2]
```diff
@@ -30,8 +31,18 @@ export const workflowWebhookHandler = async (context: WebhookContext, jiraClient
...
+		auditLogsource: "BACKFILL",
```
Comment (gxueatlassian) [inline src/github/workflow.ts:42]: SHould be `WEBHOOK`?

File: src/github/workflow.ts (hunk 1 above)
Comment (kamaksheeAtl) [reply src/github/workflow.ts:41]: ohh no yeah!

File: src/jira/client/jira-client.ts [hunk 3]
```diff
@@ -406,8 +406,21 @@ export const getJiraClient = async (
...
+				const responseData = {
+					status: response.status,
+					data:response.data
+				};
+				const reqBuildData = data?.builds[0];
+				logger?.info({ gitHubProduct }, "result ::+::",JSON.stringify(responseData));
+				if (await booleanFlag(BooleanFlags.USE_DYNAMODB_TO_PERSIST_AUDIT_LOG, jiraHost)) {
+					processAuditLogsForWorkflowSubmit({ reqBuildData, response:responseData, options, logger });
+				}
+				processAuditLogsForWorkflowSubmit({ reqBuildData, response:responseData, options, logger });
```
[the remark this answers is not in the document]
Comment (kamaksheeAtl) [reply src/jira/client/jira-client.ts:422]: I forgt to commit this change 
:P

File: src/github/workflow.ts (hunk 2 above)
Comment (kamaksheeAtl) [reply src/github/workflow.ts:42]: should be NORMAL

File: src/github/workflow.ts (hunk 1 above)
Comment (kamaksheeAtl) [reply src/github/workflow.ts:41]: should be **NORMAL**

File: src/github/workflow.ts [hunk 4]
```diff
@@ -31,7 +32,13 @@ export const workflowWebhookHandler = async (context: WebhookContext, jiraClient
 
 	logger.info({ jiraHost: jiraClient.baseURL }, `Sending workflow event to Jira`);
 
-	const jiraResponse = await jiraClient.workflow.submit(jiraPayload, payload.repository.id);
+	const jiraResponse = await jiraClient.workflow.submit(jiraPayload, payload.repository.id, {
+		preventTransitions: true,
+		operationType: "NORMAL",
+		auditLogsource: "NORMAL",
```
Comment (gxueatlassian) [inline src/github/workflow.ts:38]:
```suggestion
		auditLogsource: "WEBHOOK",
```

**Review by gxueatlassian — DISMISSED** [review]

**Review by gxueatlassian — APPROVED** [review]

**Pull request merged**